src/github/issueModel.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { Remote } from '../common/remote';
+import { GitHubRepository } from './githubRepository';
+import { IAccount, Issue, GithubItemStateEnum, IMilestone } from './interface';
+
+export class IssueModel {
+	public id: number;
+	public graphNodeId: string;
+	public number: number;
+	public title: string;
+	public html_url: string;
+	public state: GithubItemStateEnum = GithubItemStateEnum.Open;
+	public author: IAccount;
+	public assignee: IAccount;
+	public createdAt: string;
+	public updatedAt: string;
+	public milestone?: IMilestone;
+	public readonly githubRepository: GitHubRepository;
+	public readonly remote: Remote;
+	public item: Issue;
+	public bodyHTML?: string;
+	private _inDraftMode: boolean = false;
+	private _onDidChangeDraftMode: vscode.EventEmitter<boolean> = new vscode.EventEmitter<boolean>();
+	public onDidChangeDraftMode = this._onDidChangeDraftMode.event;
+
+	constructor(githubRepository: GitHubRepository, remote: Remote, item: Issue) {
+		this.githubRepository = githubRepository;
+		this.remote = remote;
+		this.item = item;
+		this.update(item);
+	}
+
+	public get isOpen(): boolean {
+		return this.state === GithubItemStateEnum.Open;
+	}
+
+	public get userAvatar(): string | undefined {
+		if (this.item) {
+			return this.item.user.avatarUrl;
+		}
+
+		return undefined;
+	}
+
+	public get userAvatarUri(): vscode.Uri | undefined {
+		if (this.item) {
+			const key = this.userAvatar;
+			if (key) {
+				const uri = vscode.Uri.parse(`${key}&s=${64}`);
+
+				// hack, to ensure queries are not wrongly encoded.
+				const originalToStringFn = uri.toString;
+				uri.toString = function (skipEncoding?: boolean | undefined) {
+					return originalToStringFn.call(uri, true);
+				};
+
+				return uri;
+			}
+		}
+
+		return undefined;
+	}
+
+	public get inDraftMode(): boolean {
```

Comment:
Sorry, I missed leaving a comment on this earlier 😱 

I think this should be moved to the PullRequestModel since only pull requests have a notion of being in draft mode. (Draft mode is overloaded, in this case it means in the middle or writing a PR review.)

Reply:
Moved all draft mode members into PullRequestModel.